### PR TITLE
Add retryCodes option in HttpMiddlewareOptions

### PR DESCRIPTION
**Pull request by ajimae**
#### Summary

- add retryCodes option to HttpMiddlewareOptions
- add unit test
- add documentation
- add flow types
- remove deprecated options from `.flowconfig`

#### Description

This adds options to allow customers retry requests on a given status (error) codes.

- Tests
  - [x] Unit
- [x] Documentation

#### Related Issue:

https://github.com/commercetools/nodejs/issues/1743

### CODE REVIEW

**Comment by lojzatran** [comment]
Hi @ajimae I don't think this retry is enough. As I wrote in the issue (https://github.com/commercetools/nodejs/issues/1743#issuecomment-1028794106), it should be able to retry also at network errors OR at 4xx errors if it contains a particular error message. Also we should be able to define the retry rules. Basically it should work the same as in the old sphere-node-sdk (https://github.com/commercetools/sphere-node-sdk/blob/master/src/coffee/repeater-task-queue.coffee#L7).

File: docs/sdk/api/sdkMiddlewareHttp.md [hunk 1]
```diff
@@ -67,6 +68,7 @@ const client = createClient({
       includeOriginalRequest: true,
       maskSensitiveHeaderData: true,
       enableRetry: true,
+      retryCodes: [504, 'ETIMEDOUT', 'ECONNREFUSED', 503],
```
Comment (jenschude) [inline docs/sdk/api/sdkMiddlewareHttp.md:71]: I would have made the retryCodes a property of the retryConfig. And use a default value array with the 503 status code like:

```
retryConfig: {
   retryCodes: [503],
  ...
}
```

File: packages/sdk-middleware-http/src/http.js [hunk 2]
```diff
@@ -59,6 +59,7 @@ export default function createHttpMiddleware({
   maskSensitiveHeaderData = true,
   enableRetry,
   timeout,
+  retryCodes = [],
```
Comment (jenschude) [inline packages/sdk-middleware-http/src/http.js:62]:
```
retryConfig: {
   retryCodes: [503],
  ...
}
```

**Comment by lojzatran** [comment]
Hi @ajimae do you have an estimation when this PR will be merged and released? Thank you.

**Comment by ajimae** [comment]
> Hi @ajimae do you have an estimation when this PR will be merged and released? Thank you.

I really can't say because we need @jenschude to approve.

**Review by jenschude — APPROVED** [review]

**Pull request merged**